File: tasks/implementations/python2/tasks/csv_parsing.py
import csv

from tasks.compat import emit, join_path, open_text


def run(size, fixtures_root):
    path = join_path(fixtures_root, 'generated', 'csv', '%s.csv' % size)
    total = 0
    handle = open_text(path)
    try:
        reader = csv.DictReader(handle)
        for row in reader:
            total += int(row['value_a']) + int(row['value_b'])
    finally:
        handle.close()
    emit(total)

File: tasks/implementations/go/tasks/csv_parsing.go
package main

import (
    "encoding/csv"
    "fmt"
    "os"
    "path/filepath"
    "strconv"
)

func RunCSVParsing(size string, fixturesRoot string) {
    file, _ := os.Open(filepath.Join(fixturesRoot, "generated", "csv", size+".csv"))
    defer file.Close()
    reader := csv.NewReader(file)
    rows, _ := reader.ReadAll()
    total := 0
    for _, row := range rows[1:] {
        a, _ := strconv.Atoi(row[1])
        b, _ := strconv.Atoi(row[2])
        total += a + b
    }
    fmt.Println(total)
}
